modules/common/src/DatasetInfoPluginManager.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\common;
+
+use Drupal\common\Annotation\DatasetInfoPlugin;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Plugin\DefaultPluginManager;
+
+/**
+ * DatasetInfo plugin manager.
+ */
+final class DatasetInfoPluginManager extends DefaultPluginManager {
+
+  /**
+   * Constructs the object.
+   */
+  public function __construct(\Traversable $namespaces, CacheBackendInterface $cache_backend, ModuleHandlerInterface $module_handler) {
+    parent::__construct('Plugin/DatasetInfo', $namespaces, $module_handler, DatasetInfoPluginInterface::class, DatasetInfoPlugin::class);
+    $this->alterInfo('dataset_info_info');
```

Comment:
Q: Should this be 'dataset_info' instead of 'dataset_info_info'?

Reply:
This is a weird one, I don't think we are really ever going to use an alter hook on this stuff, but it seems to be included in plugins by convention. dataset_info is the plugin type and _info is the standard suffix for an info_alter hook. So, this is weird but i think it's correct.